Products/ZenEvents/TrapFilter.py

```diff
@@ -0,0 +1,471 @@
+        @return: Returns None on success, or an error message on failure
+        @rtype: string
+        """
+        if len(remainingTokens) > 1:
+            return "Too many fields found; at most 3 fields allowed for V2 filters"
+
+        oid = ""
+        if len(remainingTokens) > 0:
+            oid = remainingTokens[0].strip(".")
+        result = self._validateOID(oid)
+        if result:
+            return "'%s' is not a valid OID: %s" % (oid, result)
+
+        filterDef = V2FilterDefinition(lineNumber, action, oid)
+
+        filtersByLevel = self._v2Filters.get(filterDef.levels(), None)
+        if filtersByLevel == None:
+            filtersByLevel = {oid: filterDef}
+            self._v2Filters[filterDef.levels()] = filtersByLevel
+        elif oid not in filtersByLevel:
+            filtersByLevel[oid] = filterDef
+        else:
+            previousDefinition = filtersByLevel[oid]
+            return "OID '%s' conflicts with previous definition at line %d" % (oid, previousDefinition.lineNumber)
+        return None
+
+    def _validateOID(self, oid):
+        """
+        Simplistic SNMP OID validation. Not trying to enforce some RFC spec -
+        just weed out some of the more obvious mistakes
+        """
+        if oid == "*":
+            return None
+
+        if not oid:
+            return "Empty OID is invalid"
+
+        validChars = set('0123456789.*')
+        if not all((char in validChars) for char in oid):
+            return "Invalid character found; only digits, '.' and '*' allowed"
+
+        globCount = oid.count("*")
+        if globCount > 1 or oid.startswith(".*") or (globCount == 1 and not oid.endswith(".*")):
+            return "When using '*', only a single '*' at the end of OID is allowed"
+
+        if ".." in oid:
+            return "Consecutive '.'s not allowed"
+
+        if not "." in oid:
+            return "At least one '.' required"
+        return None
+
+    def _read_filters(self):
+        fileName = self._daemon.options.trapFilterFile
+        if fileName:
+            path = zenPath('etc', fileName)
+            if os.path.exists(path):
+                with open(path) as filterDefinitionFile:
+                    lineNumber = 0
+                    for line in filterDefinitionFile:
+                        lineNumber += 1
+                        if line.startswith('#'):
+                            continue
+
+                        # skip blank lines
+                        line = line.strip()
+                        if not line:
+                            continue;
+
+                        errorMessage = self._parseFilterDefinition(line, lineNumber)
+                        if errorMessage:
+                            log.error("Failed to parse filter definition file %s at line %d: %s", format(path), lineNumber, errorMessage)
+                            log.error("Exiting due to invalid filter definition file")
+                            sys.exit(1)
```

Comment:
you can call stop in the init call back and set an exit code there.  Might throw an exception with message here.